src/main/java/com/cairn/waypoint/dashboard/endpoints/homeworkresponse/dto/AddHomeworkResponseEndpoint.java

```diff
@@ -0,0 +1,133 @@
+            security = @SecurityRequirement(name = "oAuth2JwtBearer"),
+            responses = {
+                    @ApiResponse(responseCode = "201",
+                            description = "Created - Homework response creation was successful",
+                            content = {@Content(mediaType = "application/json",
+                                    schema = @Schema(implementation = HomeworkResponseDto.class))}),
+                    @ApiResponse(responseCode = "400", description = "Bad Request",
+                            content = {@Content(schema = @Schema(hidden = true))}),
+                    @ApiResponse(responseCode = "404", description = "Not Found",
+                            content = {@Content(mediaType = "application/json",
+                                    schema = @Schema(implementation = ErrorMessage.class))}),
+                    @ApiResponse(responseCode = "409", description = "Conflict",
+                            content = {@Content(mediaType = "application/json",
+                                    schema = @Schema(implementation = ErrorMessage.class))}),
+            })
+    public ResponseEntity<?> addHomeworkResponse(
+            @RequestBody AddHomeworkQuestionResponseDto addHomeworkResponseDto, Principal principal) {
+
+        log.info("Creating homework response for question ID [{}]",
+                addHomeworkResponseDto.getQuestionId());
+
+        Optional<HomeworkQuestion> homeworkQuestionOpt =
+                homeworkQuestionDataService.getHomeworkQuestionById(addHomeworkResponseDto.getQuestionId());
+
+        if (homeworkQuestionOpt.isEmpty()) {
+            return generateFailureResponse("Homework question with ID [" +
+                            addHomeworkResponseDto.getQuestionId() + "] does not exist",
+                    HttpStatus.NOT_FOUND);
+        }
+        Optional<HomeworkCategory> categoryOpt = homeworkCategoryRepository.findById(addHomeworkResponseDto.getCategoryId());
+
+
+        if (categoryOpt.isEmpty()) {
+            return generateFailureResponse("Category with ID [" +
+                            addHomeworkResponseDto.getCategoryId() + "] does not exist",
+                    HttpStatus.NOT_FOUND);
+        }
+        HomeworkResponse homeworkResponse = HomeworkResponse.builder()
+                .response(addHomeworkResponseDto.getResponse())
+                .homeworkQuestion(homeworkQuestionOpt.get())
+                .userId(addHomeworkResponseDto.getUserId())
+                .modifiedBy(principal.getName())
+                .category(categoryOpt.get())
+                .updated(LocalDateTime.now())
+                .ordinalIndex(2)
+                .fileGuid(addHomeworkResponseDto.getFileGuid())
+                .active(true)
+                .build();
+
+        HomeworkResponse createdResponse = homeworkResponseDataService.saveHomeworkResponse(homeworkResponse);
+
+        HomeworkResponseDto responseDto = HomeworkResponseDto.builder()
```

Comment:
So note that each of these could be a single mapstruct call as opposed to the builder paradigm. This could be a  refactor for later but note there is a more concise way to construct these DTOs